rundeckapp/grails-app/domain/rundeck/ScheduledExecution.groovy

```diff
@@ -564,33 +559,65 @@ class ScheduledExecution extends ExecutionContext implements EmbeddedJsonData {
         if(data.notification){
             def nots=[]
             data.notification.keySet().findAll{it.startsWith('on')}.each{ name->
-                if(data.notification[name]){
+                def notifications = data.notification[name]
+                if(notifications){
                         //support for built-in notification types
-                    ['urls','email'].each{ subkey->
-                        if(data.notification[name][subkey]){
-                            Map notificationData = [
-                                    format    : data.notification[name]["format"],
-                                    httpMethod: data.notification[name]["httpMethod"],
-                                    (subkey)  : data.notification[name][subkey]
-                            ]
-
-                            nots << Notification.fromMap(name, notificationData)
+                    if(notifications instanceof ArrayList){
```

Comment:
is this if really necessary? is it possible that notifications is not an ArrayList?